admin/app/components/user-detail-personal-information.js

```diff
@@ -0,0 +1,110 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import Object, { action } from '@ember/object';
+import { validator, buildValidations } from 'ember-cp-validations';
+import { getOwner } from '@ember/application';
+import { inject as service } from '@ember/service';
+
+const Validations = buildValidations({
+  firstName: {
+    validators: [
+      validator('presence', {
+        presence: true,
+        ignoreBlank: true,
+        message: 'Le prénom ne peut pas être vide.'
+      }),
+      validator('length', {
+        min: 1,
+        max: 255,
+        message: 'La longueur du prénom ne doit pas excéder 255 caractères.'
+      })
+    ]
+  },
+  lastName: {
+    validators: [
+      validator('presence', {
+        presence: true,
+        ignoreBlank: true,
+        message: 'Le nom ne peut pas être vide.'
+      }),
+      validator('length', {
+        min: 0,
+        max: 255,
+        message: 'La longueur du nom ne doit pas excéder 255 caractères.'
+      })
+    ]
+  },
+  email: {
+    validators: [
+      validator('presence', {
+        presence: true,
+        ignoreBlank: true,
+        message: 'L\'e-mail ne peut pas être vide.'
+      }),
+      validator('length', {
+        min: 0,
+        max: 255,
+        message: 'La longueur de l\'email ne doit pas excéder 255 caractères.'
+      })
+    ]
+  },
+});
+
+class Form extends Object.extend(Validations) {
+  @tracked firstName;
+  @tracked lastName;
+  @tracked email;
+}
+
+export default class UserDetailPersonalInformationComponent extends Component {
+
+  @tracked isEditionMode = false;
+  @service notifications;
+
+  constructor() {
+    super(...arguments);
+    this.form = Form.create(getOwner(this).ownerInjection());
+  }
+
+  @action
+  changeEditionMode(event) {
+    event.preventDefault();
+    this._initForm();
+    this.isEditionMode = !this.isEditionMode;
+  }
+
+  @action
+  cancelEdit() {
+    this._initForm();
+    this.args.user.rollbackAttributes();
+    this.isEditionMode = true;
+  }
+
+  @action
+  async updateUserDetails(event) {
+    event.preventDefault();
+
+    const { validations } = await this.form.validate();
+    if (!validations.isValid) {
+      return;
+    }
+    this.args.user.set('firstName', this.form.firstName.trim());
```

Comment:
on doit pouvoir faire 
```suggestion
    this.args.user.firstName' = this.form.firstName.trim();
```

Reply:
De mémoire, pour l'édition d'une organization, avec @MathieuGilet on a utilisé le setter car on avait une erreur concernant `@tracker`.

Reply:
Oui, la syntaxe que tu proposes fonctionnait parfaitement et sans erreurs "en vrai" mais on a eu un problème lors de l'exécution du test `it should submit the form if there is no error` (organization-information-section-test.js). Certainement, une mauvaise injection de l'organisation dans le test mais après avoir cherché nous n'avons pas trouvé de solution. Si tu as une idée du problème on est preneur.

Reply:
Suggestion modifiée et les tests passent. On peut voir pourquoi ca passe pas  dans le détail d'organisation au point tech afin d'uniformiser la manière de valider les inputs.